src/angularComponent/ngcOmniboxController.js

```diff
@@ -90,13 +72,34 @@ export default class NgcOmniboxController {
    * @returns {Number} -- Index of the newly highlighted item
    */
   highlightPrevious() {
-    if (this.highlightedIndex > 0) {
-      this.highlightedIndex--;
+    let newIndex = this.highlightedIndex;
+
+    if (newIndex > 0) {
+      newIndex--;
+    } else {
+      newIndex = this._suggestionsUiList.length - 1;
+    }
+
+    // Protect against an infinite loop in cases where all items are disabled
+    if (this.startHighlightIndex === newIndex) {
```

Comment:
Instead of attaching `startHighlightIndex` to `this` and then needing to clear it later, can we just pass the startIndex in as a param to recursive calls of this method?